### ISSUE TITLE
Add explanation about attributions

### ISSUE THREAD

**Issue by iriberri** [CartoDB/docs#254]
Here: http://docs.cartodb.com/faqs.html#how-to-print-maps-in-cartodb

**Comment by csobier** [comment]
Closing, I mentioned editing metadata and provided link to full related procedure for the *How to Print Maps* question (in PR #337).

**Comment by iriberri** [comment]
Hey, notice "attributes" is not "attributions". Actually the new export image adds the attributions directly, so this issue is already out of date and we might just ignore all about attributions in the Export Image scope, but I don't think the naming is the perfect one. What do you think?

**Comment by csobier** [comment]
So it is not a requirement to edit the metadata for an image anymore?  It is done automatically?  If so, I need to apply some minor updates.

**Comment by iriberri** [comment]
What do you mean by "edit the metadata for an image"? :-)

Previously, when exporting an image, the attributions of the map such as "Tiles from Open Street Map data, CartoDB attribution" weren't exported, but now they're:

![image](https://cloud.githubusercontent.com/assets/1730320/9635504/0a46e6a8-5197-11e5-99d8-74cc43ac81c8.png)

Previously, we asked the users to add the "Tiles from Open Street Map data, CartoDB attribution"  attributions in whichever article or site or wherever they were going to use the exported image, but now that does no longer apply for the Export Image tool that the Editor offers.

Using the static maps API directly (something out of the editor) still has the old behaviour where no overlays such as attributions are added, but inside the editor this is already out.

**Comment by csobier** [comment]
Okay, now you have lost me... so is the requirement now that when using the Maps API for creating Static Maps, there is an attributions parameter that should be added?  If so, what is that parameter and what are examples of some values?

Maps API / Static Maps?
http://docs.cartodb.com/cartodb-platform/maps-api.html#static-maps-api

**Comment by iriberri** [comment]
Let me rephrase it again:

The attributions such as the ones of the basemaps or the CartoDB one *must* appear in any map created with the platform.

We developed the Static Maps API (the one of the link you copied) some months ago in order to create images from maps (previously this wasn't possible, you could only take a screenshot of it). 

This service was able to export the map but it's only able to export the basemap and data tiles, meaning that any other overlay in the map, such as a legend, a box element, or the attributions message itself won't appear.

As these attributions *must* appear in the maps, we needed to make sure (or at least try) that if users exported the map as an image they will mention when they publish that the map was created with CartoDB technology + it is using OSM data in the basemap (in case it was). 

In this situation, the previous "Export image" window was a small modal with a button to obtain the IMG directly as a download, and this modal window said something like: "Hey! don't forget to mention the attributions of the map when publishing the image!"

This is where this issue took place. Actually, the issue is not descriptive itself because I talked with Hayley about this and I just assigned her the issue because she knew what to do at that moment.
 
Now the new Export Image tool in CartoDB does no longer work that way, and it already adds the attribution element (as in the map I pasted above), so saying in the documentation that the attributions must be written by the user in his blog if he posted the map (random example) would no longer make sense. Probably, if I had seen this issue before I'd just closed it.

As an unrelated thing to this issue (so I don't expect nothing being done here) I wanted to explain that the Static Maps API that we used previously exists (and it's under development) so if any user uses it to retrieve images, the obligation of adding the attributions still applies, but this would have nothing to do with the Export Image button in the editor.

HTH

**Comment by csobier** [comment]
Okay, I am reopening this issue. I think the documentation requirement still applies (when using the API only).  We should add steps for how to add attributions in the Maps API/Static Maps section.  I'm adding a To Do list of related tasks for myself :-)

To Do:
- [ ] Create steps for how to add attributions when using the Static Maps API
- [ ] Update FAQ / Print Map answer to include both the automatic (Editor) and manual (API) methods and requirements for printing maps.

**Comment by iriberri** [comment]
This seems great to me. What do you plan to cover in "Create steps for how to add attributions when using the Static Maps API"?

**Comment by csobier** [comment]
Is there a targeted production milestone for Static Maps API functionality?  If so, let's add that milestone to this issue (I am still trying to figure out how to automatically sync this repo with the CartoDB mgmt milestones so that I can assign and prioritize issues).

**Comment by iriberri** [comment]
I don't think there's something else that what you can find in the general roadmap here: https://github.com/CartoDB/cartodb/milestones

**Comment by csobier** [comment]
@iriberri Can you confirm that this is the step `Image.write(attributes)` required to add attributions for a static image?  (I do not see any "attribution" parameters in any of the API docs)

http://docs.cartodb.com/cartodb-platform/cartodb-js.html#imagewriteattributes

**Comment by iriberri** [comment]
@csobier, when I'm referring to Attributions in the whole issue I'm referring to a user writing by hand a line of text in which the attributions (the ownership) of the basemaps is clear.

**Comment by csobier** [comment]
Where (and how) is this ownership information (attributions) applied for static map images?

**Comment by iriberri** [comment]
Let me try to explain it again:

When you export a map from CartoDB and we don't include the attributions for the basemaps/other attributions that might have been set up in the map [this happens when using the Static Maps API right now], the user that gets the exported image *must* mention these attributions whenever they're publishing the image out there.

Something similar is explained at the bottom of this site:

https://cartodb.com/basemaps
